connman/vpn/plugins/wireguard.c

```diff
@@ -798,6 +923,9 @@ static int disconnect(struct vpn_provider *provider, int err)
 	if (info->reresolve_id > 0)
 		g_source_remove(info->reresolve_id);
 
+	if (info->route_setup_id > 0)
+		g_source_remove(info->route_setup_id);
```

Comment:
Does it matter that stale ids are left in place?
Judging by what happens at run_dns_remove() before / after disconnect() call... bookkeeping looks fishy.

Reply:
That case in `run_dns_remove()` is because disconnect happens from  within the plugin, and the `reresolve_id` is not going to be set. Also, in `disconnect()` the preparation for the plugin burial is started and when the`wg_died()` is called as a result, the struct holding these id's is free'd so it would be unnecessary to zero these out here.

Reply:
to me, for this one timer id run_dns_reresolve(), looks like it does

```
        if (info->reresolve_id)
                g_source_remove(info->reresolve_id);
        if (info->reresolve_id > 0)
                g_source_remove(info->reresolve_id);
        info->reresolve_id = 0;

```
it is not going to cause crashes / like but... oh well, it is what it is

Reply:
Ok, I forgot to use Simo-English-Simo dictionary. Yeah, the order of the zeroing was faulty, fixed.